### PR TITLE
DD-1637 Add age option to deposit-createreport script

### PR DESCRIPTION
**Pull request by aliassheikh**
Fixes DD-1637 age optie voor deposit-create-report

# Description of changes
added to `~/git/dans-core-systems/modules/dans-datastation-tools/src/datastation/deposit_create_report.py`
`add_argument('-a', '--age', dest='age', help='the number of days before today ')` 
When it is given in the command line, it will set the startdate argument to asked date, if it is an integer,

# How to test
For example, in vagrant ssh:
`python3  /vagrant/modules/dans-datastation-tools/src/datastation/deposit_create_report.py --help`
# Related PRs

(Add links)

*

# Notify

@DANS-KNAW/core-systems

### CODE REVIEW

File: src/datastation/deposit_create_report.py [hunk 1]
```diff
@@ -46,7 +48,11 @@ def main():
     parser.add_argument('-o', '--output-file', dest='output_file', default='-',
                         help='the file to write the output and send recipient to a to or - for stdout')
     parser.add_argument('-e', '--enddate', dest='enddate', help='Filter until the record creation of this date')
-    parser.add_argument('-s', '--startdate', dest='startdate', help='Filter from the record creation of this date')
+
+    group = parser.add_mutually_exclusive_group(required=False)
+    group.add_argument('-s', '--startdate', dest='startdate', help='Filter from the record creation of this date')
+    group.add_argument('-a', '--age', dest='age', help='the number of days before today ')
```
Comment (jo-pol) [inline src/datastation/deposit_create_report.py:54]:
```suggestion
    group.add_argument('-a', '--age', dest='age', help='Filter from record creation not older than a number of days before today ')
```

File: src/datastation/deposit_create_report.py [hunk 2]
```diff
@@ -56,6 +62,9 @@ def main():
     parser.add_argument('--bcc-email-to', dest='bcc_email_to', help='will be sent only if email-to is defined')
     args = parser.parse_args()
 
+    if args.age is not None and str(args.age).isnumeric(): # Note: args is a Namespace object
```
Comment (janvanmansum) [inline src/datastation/deposit_create_report.py:65]: You can just tell argparse it should be in an int at line 54.

File: src/datastation/deposit_create_report.py [hunk 3]
```diff
@@ -46,7 +48,11 @@ def main():
     parser.add_argument('-o', '--output-file', dest='output_file', default='-',
                         help='the file to write the output and send recipient to a to or - for stdout')
     parser.add_argument('-e', '--enddate', dest='enddate', help='Filter until the record creation of this date')
-    parser.add_argument('-s', '--startdate', dest='startdate', help='Filter from the record creation of this date')
+
+    group = parser.add_mutually_exclusive_group(required=False)
+    group.add_argument('-s', '--startdate', dest='startdate', help='Filter from the record creation of this date')
+    group.add_argument('-a', '--age', dest='age', help='Filter from record creation not older than a number of days before today ')
```
Comment (janvanmansum) [inline src/datastation/deposit_create_report.py:54]:
```suggestion
    group.add_argument('-a', '--age', dest='age', type=int,  help='Filter from record creation not older than a number of days before today ')
```

File: src/datastation/deposit_create_report.py (hunk 2 above)
Comment (janvanmansum) [reply src/datastation/deposit_create_report.py:65]: See: https://docs.python.org/3/library/argparse.html#example

**Review by janvanmansum — CHANGES_REQUESTED** [review]

**Pull request merged**